src/App.tsx

```diff
@@ -1,61 +1,135 @@
+import React, { useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './components/Types/Todo';
+import { User } from './components/Types/User';
+import 'bulma/css/bulma.css';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function findUserById(userId: number): User | null {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+function findUserByName(userName: string): User | null {
+  return usersFromServer.find(user => userName === user.name) || null;
+}
+
+const preparedTodos: Todo[] = todosFromServer.map(todo => {
+  return {
+    ...todo,
+    user: findUserById(todo.userId),
+  };
+});
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState('');
+  const [todos, setTodos] = useState(preparedTodos);
+  const [isErrorOnUserSelect, setErrorOnUserSelect] = useState(false);
+  const [isErrorOnTitleInput, setErrorOnTitleInput] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.currentTarget.value);
+    setErrorOnTitleInput(false);
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUser(event.currentTarget.value);
+    setErrorOnUserSelect(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setErrorOnTitleInput(!title.trim());
+    setErrorOnUserSelect(!selectedUser);
+
+    if (title.trim() === '' || !selectedUser) {
+      return;
+    }
+
+    const userToAdd = findUserByName(selectedUser);
+
+    setTodos(current => {
+      const maxTodoId = Math.max(...current.map(todo => todo.id));
+
+      return [
+        ...current,
+        {
+          id: maxTodoId + 1,
+          title,
+          completed: false,
+          userId: userToAdd ? userToAdd.id : null,
+          user: userToAdd,
+        },
```

Comment:
create variable to increase code readable

Reply:
fixed it